### ISSUE TITLE
Issue on `classy-C4`

### ISSUE THREAD

**Issue by The-Lum** [james-gadrow-kr/classy-c4#1]
Hello @james-gadrow-kr,

It seems that there are some issues on the `classy-c4` lib.

Observed on:
- plantuml/plantuml-stdlib#56

Here is a minimal example:
```puml
@startuml
!include <classy-c4/container>
!include <classy-c4/person>
!include <classy-c4/system>
@enduml
```
>[![](https://www.planttext.com/api/plantuml/svg/SoWkIImgAStDuLBCp4lEAKr9LR19paaiBgxKJJRHJyxFAqdCp4ijiiCgNP1QL9oVXrskkBAu93NNZiiXDIy5Q3m0)](https://www.planttext.com/?text=SoWkIImgAStDuLBCp4lEAKr9LR19paaiBgxKJJRHJyxFAqdCp4ijiiCgNP1QL9oVXrskkBAu93NNZiiXDIy5Q3m0) 

> Assertion error : [ERROR] Class already defined.

- Is it normal or not?
- What is the normal behavior or usage?

_(FYI @anb0s)_
Regards,
Th.

**Comment by james-gadrow-kr** [comment]
Thanks for bringing this to my attention! From that discussion, seems like this has been in place for a very long time but this is the first I've seen anything about it!

So, yes, I would have expected that to not provide you with an error. I'll try and find some time to get this figured out and a fix committed.

I'll probably be submitting some modifications to classy core as well because I don't know why I didn't think to indicate *which* class was already defined. Would help a ton with the debugging. ;)

**Comment by james-gadrow-kr** [comment]
I have added implementation to correct this issue. This was being caused by files that were being included more than once. I am surprised this was necessary as the preprocessor documentation indicates that !include is supposed to only include a file one time. Regardless, I have wrapped files in a conditional now which prevents this issue from occurring.

I also added the minor change above of saying *which* class was already defined to assist with debugging in the future.

I have submitted these new versions to the plantuml maintainers as fixes for this issue. As soon as they accept it, this should fix the issue.
